### ISSUE TITLE
pytest -v Issues

### ISSUE THREAD

**Issue by dbl001** [allenai/allennlp#284]
I installed python 3.6 into an Anaconda environment installed all the requirements.txt and requirements_test.txt packages, pytorch, etc.

When I run pytest -v, the tests are failing.   
It is accessing the python 2.7 packages (not the python 3.6 ones):

```
(py36) David-Laxers-MacBook-Pro:allennlp davidlaxer$ pytest -v
============================= test session starts ==============================
platform darwin -- Python 2.7.13, pytest-3.1.1, py-1.4.33, pluggy-0.4.0 -- /Users/davidlaxer/anaconda/bin/python
cachedir: .cache
rootdir: /Users/davidlaxer/allennlp, inifile: pytest.ini
collected 0 items / 68 errors 

==================================== ERRORS ====================================
___________________ ERROR collecting tests/notebooks_test.py ___________________
../anaconda/lib/python2.7/site-packages/_pytest/python.py:408: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
../anaconda/lib/python2.7/site-packages/py/_path/local.py:662: in pyimport
    __import__(modname)
E     File "/Users/davidlaxer/allennlp/tests/notebooks_test.py", line 17
E       def execute_notebook(notebook_path: str):
E                                         ^
E   SyntaxError: invalid syntax
[...]
ImportError while importing test module '/Users/davidlaxer/allennlp/tests/training/metrics/span_based_f1_measure_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tests/training/metrics/span_based_f1_measure_test.py:2: in <module>
    import torch
E   ImportError: No module named torch
!!!!!!!!!!!!!!!!!!! Interrupted: 68 errors during collection !!!!!!!!!!!!!!!!!!!
=========================== 68 error in 5.49 seconds ===========================
```

**Comment by matt-gardner** [comment]
You probably need to put the anaconda version of python first on your $PATH.

**Comment by dbl001** [comment]
(py36) David-Laxers-MacBook-Pro:allennlp davidlaxer$ env | grep $PATH
PATH=/Users/davidlaxer/anaconda/envs/py36/bin:/Users/davidlaxer/anaconda/bin:/Users/davidlaxer/.sdkman/candidates/springboot/current/bin:/Users/davidlaxer/.jenv/shims:/Users/davidlaxer/.jenv/bin:/Developer/NVIDIA/CUDA-6.5/bin:/usr/local/lib/erlang/erts-5.9.1/bin:/Users/davidlaxer/anaconda/bin:/Users/davidlaxer/anaconda/bin/bin:/opt/local/Library/Frameworks/Python.framework/Versions/2.7/bin:/usr/bin:/opt/local/bin:/opt/local/sbin:/usr/local/lib/erlang/erts-5.9.1/bin:/Users/davidlaxer/anaconda/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/usr/local/CrossPack-AVR/bin:/usr/local/git/bin:/usr/texbin::/opt/local/bin/scala/bin::/usr/local/hadoop/bin:/usr/local/hadoop/sbin:/Users/davidlaxer/pig-0.13.0-src/bin:/Users/davidlaxer/pig-0.13.0-src/sbin:/Users/davidlaxer/Downloads/apache-maven-3.2.1/bin:/Users/davidlaxer/anaconda/bin/bin::/usr/local/hadoop/bin:/usr/local/hadoop/sbin:/Users/davidlaxer/.rvm/bin:/Library/Frameworks/JRuby.framework/Versions/Current/bin:/usr/local/gradle-source-build/bin:/Users/davidlaxer/bin:/Users/davidlaxer/storm/bin:/usr/bin:/usr/sbin:/Users/davidlaxer/spark/bin:/Library/PostgreSQL/9.4/bin:/usr/local/mysql/bin

**Comment by matt-gardner** [comment]
`which pytest` should give you something inside of an anaconda directory.  If it's not there, you need to figure out how to get it there.

**Comment by dbl001** [comment]
$ which pytest
/Users/davidlaxer/anaconda/envs/py36/bin/pytest

**Comment by dbl001** [comment]
This worked (one test failed):

$ /Users/davidlaxer/anaconda/envs/py36/bin/pytest -v

…

=================================== FAILURES ===================================
___________________ TestSrlPredictor.test_uses_named_inputs ____________________

self = <tests.service.predictors.srl_test.TestSrlPredictor testMethod=test_uses_named_inputs>

    def test_uses_named_inputs(self):
        inputs = {
                "sentence": "The squirrel wrote a unit test to make sure its nuts worked as designed."
        }
    
        archive = load_archive('tests/fixtures/srl/serialization/model.tar.gz')
        predictor = Predictor.from_archive(archive)
    
        result = predictor.predict_json(inputs)
    
        # TODO(joelgrus): update this when you figure out the result format
        verbs = result.get("verbs")
    
        assert verbs is not None
    
>       assert any(v["verb"] == "wrote" for v in verbs)
E       AssertionError: assert False
E        +  where False = any(<generator object TestSrlPredictor.test_uses_named_inputs.<locals>.<genexpr> at 0x111410db0>)

tests/service/predictors/srl_test.py:24: AssertionError
----------------------------- Captured stdout call -----------------------------

    Warning: no model found for 'en'

    Only loading the 'en' tokenizer.

----------------------------- Captured stderr call -----------------------------
2017-09-09 12:33:05,798 - INFO - allennlp.models.archival - extracting archive file tests/fixtures/srl/serialization/model.tar.gz to temp dir /var/folders/nj/nphdkhyj6s1dttb0pd9zb2wc0000gn/T/tmplh2i368e
2017-09-09 12:33:05,851 - INFO - allennlp.data.vocabulary - Loading token dictionary from /var/folders/nj/nphdkhyj6s1dttb0pd9zb2wc0000gn/T/tmplh2i368e/vocabulary.
2017-09-09 12:33:05,854 - PARAM - allennlp.common.params - model.type = srl
2017-09-09 12:33:05,854 - PARAM - allennlp.common.params - model.text_field_embedder.type = basic
2017-09-09 12:33:05,854 - PARAM - allennlp.common.params - model.text_field_embedder.tokens.type = embedding
2017-09-09 12:33:05,855 - PARAM - allennlp.common.params - model.text_field_embedder.tokens.num_embeddings = None
2017-09-09 12:33:05,855 - PARAM - allennlp.common.params - model.text_field_embedder.tokens.vocab_namespace = tokens
2017-09-09 12:33:05,856 - PARAM - allennlp.common.params - model.text_field_embedder.tokens.embedding_dim = 100
2017-09-09 12:33:05,856 - PARAM - allennlp.common.params - model.text_field_embedder.tokens.pretrained_file = None
2017-09-09 12:33:05,856 - PARAM - allennlp.common.params - model.text_field_embedder.tokens.projection_dim = None
2017-09-09 12:33:05,857 - PARAM - allennlp.common.params - model.text_field_embedder.tokens.trainable = True
2017-09-09 12:33:05,857 - PARAM - allennlp.common.params - model.text_field_embedder.tokens.padding_index = None
2017-09-09 12:33:05,857 - PARAM - allennlp.common.params - model.text_field_embedder.tokens.max_norm = None
2017-09-09 12:33:05,858 - PARAM - allennlp.common.params - model.text_field_embedder.tokens.norm_type = 2.0
2017-09-09 12:33:05,858 - PARAM - allennlp.common.params - model.text_field_embedder.tokens.scale_grad_by_freq = False
2017-09-09 12:33:05,858 - PARAM - allennlp.common.params - model.text_field_embedder.tokens.sparse = False
2017-09-09 12:33:05,860 - PARAM - allennlp.common.params - model.stacked_encoder.type = lstm
2017-09-09 12:33:05,860 - PARAM - allennlp.common.params - model.stacked_encoder.batch_first = True
2017-09-09 12:33:05,861 - INFO - allennlp.common.params - Converting Params object to dict; logging of default values will not occur when dictionary parameters are used subsequently.
2017-09-09 12:33:05,861 - INFO - allennlp.common.params - CURRENTLY DEFINED PARAMETERS: 
2017-09-09 12:33:05,861 - PARAM - allennlp.common.params - model.stacked_encoder.input_size = 150
2017-09-09 12:33:05,861 - PARAM - allennlp.common.params - model.stacked_encoder.hidden_size = 10
2017-09-09 12:33:05,862 - PARAM - allennlp.common.params - model.stacked_encoder.num_layers = 1
2017-09-09 12:33:05,862 - PARAM - allennlp.common.params - model.stacked_encoder.batch_first = True
2017-09-09 12:33:05,863 - PARAM - allennlp.common.params - model.binary_feature_dim = 50
2017-09-09 12:33:05,863 - PARAM - allennlp.common.params - model.initializer = []
2017-09-09 12:33:05,864 - INFO - allennlp.nn.initializers - Initializing parameters
2017-09-09 12:33:05,865 - INFO - allennlp.nn.initializers - Done initializing parameters; the following parameters are using their default initialization from their code
2017-09-09 12:33:05,865 - INFO - allennlp.nn.initializers -    binary_feature_embedding.weight
2017-09-09 12:33:05,865 - INFO - allennlp.nn.initializers -    stacked_encoder._module.bias_hh_l0
2017-09-09 12:33:05,866 - INFO - allennlp.nn.initializers -    stacked_encoder._module.bias_ih_l0
2017-09-09 12:33:05,866 - INFO - allennlp.nn.initializers -    stacked_encoder._module.weight_hh_l0
2017-09-09 12:33:05,866 - INFO - allennlp.nn.initializers -    stacked_encoder._module.weight_ih_l0
2017-09-09 12:33:05,867 - INFO - allennlp.nn.initializers -    tag_projection_layer._module.bias
2017-09-09 12:33:05,867 - INFO - allennlp.nn.initializers -    tag_projection_layer._module.weight
2017-09-09 12:33:05,867 - INFO - allennlp.nn.initializers -    text_field_embedder.token_embedder_tokens.weight
2017-09-09 12:33:05,869 - PARAM - allennlp.common.params - dataset_reader.type = srl
=============== 1 failed, 257 passed, 1 skipped in 37.18 seconds ===============
(py36) David-Laxers-MacBook-Pro:allennlp davidlaxer$

**Comment by matt-gardner** [comment]
What have you done to try solving this yourself?  This is pretty clearly an issue with your environment; I can't debug your environment for you.  I'd suggest searching on google for why pytest would be using the wrong version of python.

**Comment by matt-gardner** [comment]
For that issue, you need to download the spacy models, which is done for you in the `install_requirements.sh` script mentioned in the README.

**Comment by matt-gardner** [comment]
Not sure why you have to specify the full path to `pytest`; there appears to be something funny going on with your path somehow.

**Comment by dbl001** [comment]
On my Ubuntu 14.04 LTS EC2 instance, ‘pytest’ is NOT included in the python 3.6 environment.
But, It is included in the python 3.6 environment on OS X.

Still, investigating with the wrong version of pytest was invoked even thought the python 3.6 environment was the first one in my $PATH.
Thanks for your help!
